### PR TITLE
Screen share implementation

### PR DESCRIPTION
**Pull request by jashan1498**

### CODE REVIEW

**Review by govindmaheshwari2 — APPROVED** [review]

**Comment by ygit** [comment]
@jashan1498 can you resolve the conflicts?

File: android/src/main/kotlin/live/hms/hmssdk_flutter/Constants.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package live.hms.hmssdk_flutter
+
+class Constants {
+    companion object{
+        const val SCREEN_SHARE_INTENT_REQUEST_CODE = 192
+    }
+
+
+}
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/Constants.kt:9]: always add an empty line to the end of files

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 2]
```diff
@@ -95,6 +105,9 @@ class HmssdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
             "switch_video" -> {
                 switchVideo(call, result)
             }
+            "is_screen_share_active" -> {
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:108]: all screenshare related cases should be together, below one another
`is_screen_share_active` case should be around `start_screen_share` & `stop_screen_share`

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 3]
```diff
@@ -576,9 +595,33 @@ class HmssdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
         result.success(HMSPeerExtension.toDictionary(getLocalPeer()))
     }
 
+
+    private fun startScreenShare() {
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:599]: `startScreenShare` function should return `result.success(nil)` if successful or exception hashmap on error

File: android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt [hunk 4]
```diff
@@ -937,6 +979,19 @@ class HmssdkFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
         }
     }
 
+    fun requestScreenShare(data: Intent?) {
+        hmssdk.startScreenshare(object : HMSActionResultListener {
```
Comment (ygit) [inline android/src/main/kotlin/live/hms/hmssdk_flutter/HmssdkFlutterPlugin.kt:983]: should return `result.success(nil)` if successful or exception hashmap on error
use the existing result action listener

File: example/lib/meeting/hms_sdk_interactor.dart [hunk 5]
```diff
@@ -172,6 +176,14 @@ class HMSSDKInteractor {
     hmsSDK.muteAll();
   }
 
+  void startScreenShare() {
```
Comment (ygit) [inline example/lib/meeting/hms_sdk_interactor.dart:179]: should have a action result listener

File: example/lib/meeting/meeting_store.dart [hunk 6]
```diff
@@ -611,6 +618,16 @@ abstract class MeetingStoreBase extends ChangeNotifier
     _hmssdkInteractor.removePeer(peer, this);
   }
 
+  void startScreenShare() {
```
Comment (ygit) [inline example/lib/meeting/meeting_store.dart:621]: should have a action result listener

File: lib/src/common/platform_methods.dart [hunk 7]
```diff
@@ -121,6 +130,8 @@ extension PlatformMethodValues on PlatformMethod {
         return 'switch_video';
       case PlatformMethod.switchCamera:
         return 'switch_camera';
+      case PlatformMethod.isScreenShareActive:
```
Comment (ygit) [inline lib/src/common/platform_methods.dart:133]: all new API cases should be added to the end.

File: lib/src/common/platform_methods.dart [hunk 8]
```diff
@@ -173,6 +184,10 @@ extension PlatformMethodValues on PlatformMethod {
         return "get_room";
       case PlatformMethod.stopRtmpAndRecording:
         return "stop_rtmp_and_recording";
+      case PlatformMethod.startScreenShare:
```
Comment (ygit) [inline lib/src/common/platform_methods.dart:187]: all new API cases should be added to the end.

File: lib/src/common/platform_methods.dart [hunk 9]
```diff
@@ -214,7 +229,9 @@ extension PlatformMethodValues on PlatformMethod {
         return PlatformMethod.onReconnecting;
       case 'on_re_connected':
         return PlatformMethod.onReconnected;
-      case 'switch_audio':
+      case 'on_re_connected':
+        return PlatformMethod.isScreenShareActive;
```
Comment (ygit) [inline lib/src/common/platform_methods.dart:233]: all new API cases should be added to the end.

File: lib/src/hmssdk.dart [hunk 10]
```diff
@@ -112,6 +112,25 @@ class HMSSDK {
     );
   }
 
+  /// start screen share for the meeting
+  Future<void> startScreenShare() async {
```
Comment (ygit) [inline lib/src/hmssdk.dart:116]: should be a void function with action result listener

File: lib/src/hmssdk.dart [hunk 11]
```diff
@@ -112,6 +112,25 @@ class HMSSDK {
...
+    );
+  }
+
+  Future<bool> isScreenShareActive() async {
+    return await PlatformService.invokeMethod(
+      PlatformMethod.isScreenShareActive,
+    );
+  }
+
+  Future<void> stopScreenShare() async {
```
Comment (ygit) [inline lib/src/hmssdk.dart:128]: should be a void function with action result listener

File: pubspec.lock [hunk 12]
```diff
@@ -7,7 +7,7 @@ packages:
       name: async
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "2.8.2"
+    version: "2.8.1"
```
Comment (ygit) [inline pubspec.lock:10]: don't downgrade the pubspec dependencies

**Review by ygit — CHANGES_REQUESTED** [review]

**Pull request merged**